src/ApiCore/OperationResponse.php

```diff
@@ -133,40 +164,68 @@ public function getName()
      * Poll the server in a loop until the operation is complete.
      *
      * Return true if the operation completed, otherwise return false. If the
-     * $options['maxPollingDuration'] setting is not set (or set <= 0.0) then
+     * $options['totalPollTimeoutMillis'] setting is not set (or set <= 0.0) then
      * pollUntilComplete will continue polling until the operation completes,
      * and therefore will always return true.
      *
      * @param array $options {
      *                       Options for configuring the polling behaviour.
      *
-     *     @type float $pollingIntervalSeconds The polling interval to use, in seconds.
-     *                                          Default: 1.0
-     *     @type float $maxPollingDurationSeconds The maximum amount of time to continue polling.
-     *                                            Default: 0.0 (no maximum)
+     *     @type float $initialPollDelayMillis The initial polling interval to use, in seconds.
+     *     @type float $pollDelayMultiplier    Multiplier applied to the polling interval on each retry.
+     *     @type float $maxPollDelayMillis     The maximum polling interval to use, in seconds.
+     *     @type float $totalPollTimeoutMillis The maximum amount of time to continue polling.
      * }
      * @throws ApiException If an API call fails.
+     * @throws ValidationException
      * @return bool Indicates if the operation completed.
      */
     public function pollUntilComplete($options = [])
     {
-        $defaultPollSettings = [
-            'pollingIntervalSeconds' => $this::DEFAULT_POLLING_INTERVAL,
-            'maxPollingDurationSeconds' => 0.0,
-        ];
-        $pollSettings = array_merge($defaultPollSettings, $options);
+        $pollSettings = array_merge($this->defaultPollSettings, $options);
+
+        $currentPollDelayMillis = $pollSettings['initialPollDelayMillis'];
+        $pollDelayMultiplier = $pollSettings['pollDelayMultiplier'];
+        $maxPollDelayMillis = $pollSettings['maxPollDelayMillis'];
+        $totalPollTimeoutMillis = $pollSettings['totalPollTimeoutMillis'];
 
-        $pollingIntervalMicros = $pollSettings['pollingIntervalSeconds'] * 1000000;
-        $maxPollingDuration = $pollSettings['maxPollingDurationSeconds'];
+        $hasTotalPollTimeout = $totalPollTimeoutMillis > 0.0;
+        $endTime = $this->getCurrentTimeMillis() + $totalPollTimeoutMillis;
 
-        $hasMaxPollingDuration = $maxPollingDuration > 0.0;
-        $endTime = microtime(true) + $maxPollingDuration;
-        while (!$this->isDone() && (!$hasMaxPollingDuration || microtime(true) < $endTime)) {
-            usleep($pollingIntervalMicros);
+        while (true) {
+            if ($this->isDone()) {
+                return true;
+            }
+            if ($hasTotalPollTimeout && $this->getCurrentTimeMillis() > $endTime) {
+                return $this->isDone();
```

Comment:
Could this be added as an `elseif` to the above condition and just always return false instead of checking the return of `isDone()` again?

```php
if ($this->isDone()) {
    return true;
} elseif ($hasTotalPollTimeout && $this->getCurrentTimeMillis() > $endTime) {
    return false;
}
```

Reply:
Fixed